### DISCUSSION TITLE
Accident caused by --remove-source-files

### DISCUSSION THREAD

**Discussion by huangyingw** [RsyncProject/rsync#193] [General]
when I used --remove-source-files in following way:

````
rsync --remove-source-files source_dir/ target_dir/
````

there is a great danger: 
if I Accidentally run it with source_dir === target_dir, or source_dir is just a symbol link to target_dir, 
then both the source_dir and target_dir would be clean, empty, all the Contents under these dir would be gone.

Is there any pre-check for --remove-source-files, to prevent this case?

**Comment by WayneD** [comment]
Only in your own brain. When prepping an rsync, it's often good to use option `-ai` and `--dry-run` (`-n`) to see what's going to happen. If you pointed it at the same source/dest files, it would show that there was nothing to copy, which should be a red flag.
